config/config.go

```diff
@@ -293,3 +293,9 @@ func DifferentNamespaceConfigs(c1, c2 *pb.NamespaceConfig) bool {
 
 	return false
 }
+
+func cloneConfig(cfg *pb.ServiceConfig) *pb.ServiceConfig {
+	r, _ := Marshal(cfg)
```

Comment:
Performance implications of this aren't great. Do we care?

Reply:
I was actually thinking about the implications of updating `persister.go` to take the actual protos as arguments, rather than `readers`. We could probably look into that in a separate PR, as it would be a decent refactor.